src/sentry/workflow_engine/processors/schedule.py

```diff
@@ -119,6 +122,22 @@ def project_ids_to_process(
         # With this, we distribute cohorts across runs, but ensure we don't process them
         # too frequently or too late, while not being too dependent on number of cohorts or
         # frequency of scheduling.
+        if len(cohort_updates.values) != self.num_cohorts:
+            logger.info(
+                "%s cohort_updates.values, but num_cohorts is %s. Resetting.",
+                len(cohort_updates.values),
+                self.num_cohorts,
+                extra={"cohort_updates": cohort_updates.values},
+            )
+            # When cohort counts change, we accept that we'll be potentially running some
+            # projects a bit too early. Previous cohorts are no longer valid, so the timestamps
+            # associated with them are only accurate for setting bounds on the whole project space.
+            # But, we can still use that to avoid running projects too early by giving them all the
+            # eldest timestamp.
+            eldest = min(cohort_updates.values.values(), default=long_ago)
+            # This also ensures that if we downsize cohorts, we don't let data from now-dead cohorts
+            # linger.
+            cohort_updates.values = {co: eldest for co in range(self.num_cohorts)}
```

Comment:
Does setting all of the cohort update times to the same time make it so that you can pick any of them to run at the current time?

Does this potentially skip some projects that we should definitely run?